src/likelihoods/categorical.jl

```diff
@@ -9,8 +9,12 @@ uncertainity associated with the data follows a Categorical distribution.
 On calling, this would return a Categorical distribution with `f_i` 
 probability of `i` category.
 """
-struct CategoricalLikelihood end
+struct CategoricalLikelihood{Tl<:AbtractLink}
+    invlink::Tl
+end
 
-(l::CategoricalLikelihood)(f::AbstractVector{<:Real}) = Categorical(softmax(vcat(f, 0)))
+CategoricalLikelihood(invlink::T=SoftMaxLink()) where{T<:AbstractLink} = CategoricalLikelihood{T}(invlink)
```

Comment:
```suggestion
CategoricalLikelihood() = CategoricalLikelihood(SoftMaxLink())
```
You get the single argument version for free frm the default constructor.